### PR TITLE
[macos] add retries to brew install

### PR DESCRIPTION
**Pull request by ilia-shipitsin**
# Description

add retries to Homebrew install process

## Check list
- [ ] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images/macos/tests/Python.Tests.ps1 [hunk 1]
```diff
@@ -8,8 +8,8 @@ Describe "Python3" -Skip:($os.IsVenturaArm64) {
         "python3 --version" | Should -ReturnZeroExitCode
     }
 
-    It "Python 3 is installed under /usr/local/bin" {
-        Get-WhichTool "python3" | Should -BeLike "/usr/local/bin*"
+    It "Python 3 is installed under /usr/bin" {
```
Comment (mikhailkoliada) [inline images/macos/tests/Python.Tests.ps1:11]: we test exactly `/usr/local` here

Comment (ilia-shipitsin) [reply images/macos/tests/Python.Tests.ps1:11]: thus, this should be addressed in other way.
"python3" lives in /usr/bin 

if that's unexpected, ok, let's fix that

**Review by Alexey-Ayupov — APPROVED** [review]

**Pull request closed without merging**